src/commands/PublishCommand.js

```diff
@@ -272,7 +272,9 @@ export default class PublishCommand extends Command {
       if (this.options.canary) {
         this.logger.info("canary", "Resetting git state");
         // reset since the package.json files are changed
-        GitUtilities.checkoutChanges("packages/*/package.json", this.execOpts);
+        this.repository.packageParentDirs.forEach(packageDir => {
+          GitUtilities.checkoutChanges(`${packageDir}/*/package.json`, this.execOpts);
```

Comment:
As enticing as the `packageParentDirs` are, we still need to support some weird configs (like `"."`, ugh) for package locations. We should iterate over `this.repository.packageConfigs`, joining the filename to the glob:

```js
this.repository.packageConfigs.forEach(pkgGlob => {
  GitUtilities.checkoutChanges(`${pkgGlob}/package.json`, this.execOpts);
});
```

The test will stay the same, and the existing tests are sufficient for this feature.